app/script.js

```diff
@@ -5,15 +5,26 @@ const app = new Aragon()
 const initialState = {
   count: 0
 }
-app.store((state, event) => {
+app.store(async (state, event) => {
   if (state === null) state = initialState
 
   switch (event.event) {
     case 'Increment':
-      return { count: parseInt(state.count, 10) + parseInt(event.returnValues.step, 10) }
+      return { count: await getValue() }
     case 'Decrement':
-      return { count: parseInt(state.count, 10) - parseInt(event.returnValues.step, 10) }
+      return { count: await getValue() }
```

Comment:
I guess you could get rid of the switch, or at least join those 2 cases.

Reply:
Wanted to keep the switch so it was obvious what the pattern would be for a more complicated app :).